dart/lib/flat_buffers.dart

```diff
@@ -465,12 +465,12 @@ class Builder {
     }
   }
 
-  /// Start a new table.  Must be finished with [endTable] invocation.
-  void startTable() {
+  /// Start a new table. Must be finished with [endTable] invocation.
+  void startTable(int numFields) {
     if (_currentVTable != null) {
       throw new StateError('Inline tables are not supported.');
     }
-    _currentVTable = new _VTable();
+    _currentVTable = new _VTable(numFields);
```

Comment:
why are we allocating a new vtable on every table serialized anyway? other lang impls have an array that gets reused, and only resized when a table with a larger vtable is encountered

Reply:
hmm, that would make it faster for multi-table buffers but slower for a single-table one. Also this PR introduces a fixed-size typed array which should be much faster to access (index) than a standard expandable list, but it cannot be resized, only replaced by a different one.

Reply:
Might have a look at a table reuse later, would still prefer this merged.

Reply:
I'd think this is a big speedup for multi-table buffers, and should not matter much for single table..

But yes wasn't suggesting you do it in this PR